R/oauth-init.R

```diff
@@ -13,13 +13,20 @@
 #' @keywords internal
 init_oauth1.0 <- function(endpoint, app, permission = NULL,
                           is_interactive = interactive(),
-                          private_key = NULL) {
+                          private_key = NULL, use_oob = getOption("httr_oob_default")) {
 
+  if (isTRUE(use_oob)) {
+    stopifnot(interactive())
```

Comment:
Let's reduce the duplication between here and `init_oauth2.0` and create a new function like:

```R
can_interactive_oauth <- function(oob) {
  oob <- isTRUE(oob)
  if (!is_installed("httpuv")) {
     message("httpuv not installed, defaulting to out-of-band authentication")
     oob <- TRUE
  }

  if (oob && !interactive) {
    stop("OOB auth requires an interactive session", call. = FALSE)
  }

  oob
}
```